### PR TITLE
fix: append `async` keyword when replace async func with useCallback

### PR DESCRIPTION
**Pull request by fossamagna**
## Description

the following code

```jsx
const Component = () => {
  const myFn = async () => [];
  return <Child prop={myFn} />;
}
```

disappear `async` keyword when fixed

```jsx
import { useCallback } from 'react';

const Component = () => {
  const myFn = useCallback(() => [], []);
  return <Child prop={myFn} />;
}
```

## Checklist

- [x] **Tests**: I have created multiple test case scenarios for my changes.
- [x] **Passing Tests**: All existing and new tests are passing.
- [x] **Version Bump**: I have increased the package version number in `package.json` following the [Semantic Versioning](https://semver.org/) (SEMVER) standard.
- [x] **Focused Changes**: My code changes are focused solely on the matter described above.

## Additional Information

n/a

### CODE REVIEW

**Comment by fossamagna** [comment]
Hi @arthurgeron , Cloud you review this pull request ?

**Comment by arthurgeron** [comment]
> Hi @arthurgeron , Cloud you review this pull request ?

Hey, I was focused on Fuel Lab's Mainnet launch. Now I'll have more time to take care of this project.. Will launch a review soon.

File: __tests__/require-usecallback.test.ts [hunk 1]
```diff
@@ -259,6 +259,34 @@ describe('Rule - Require-usecallback', () =>  {
...
+        code: `
+        const Component = () => {
+          const myFn = async () => [];
+          return <Child prop={myFn} />;
+        }`,
+        output: `import { useCallback } from 'react';
+
+        const Component = () => {
+          const myFn = useCallback(async () => [], []);
+          return <Child prop={myFn} />;
```
Comment (arthurgeron) [inline __tests__/require-usecallback.test.ts:286]: I'd be good to also test how auto-fix would work for inline async methods. It's a good idea to also add that to the tests :)
```suggestion
          return <Child prop={async () => {}} />;
```

**Review by arthurgeron — CHANGES_REQUESTED** [review]
LGTM, just need to add more tests to be safe.

File: __tests__/require-usecallback.test.ts (hunk 1 above)
Comment (fossamagna) [reply __tests__/require-usecallback.test.ts:286]: @arthurgeron 
Thank you for your review.
I add a test case for inline async function.

File: __tests__/require-usecallback.test.ts [hunk 2]
```diff
@@ -259,6 +259,41 @@ describe('Rule - Require-usecallback', () =>  {
...
+        }`,
+        errors: [{ messageId: "function-usecallback-props" }],
+      },
+      {
+        code: `
+        const Component = () => {
+          return <Child prop={async () => []} />;
+        }`,
+        errors: [{ messageId: "function-usecallback-props" }],
+      },
```
Comment (arthurgeron) [inline __tests__/require-usecallback.test.ts:296]: Shouldn't the autofix create a prop like it does for the other cases?
```suggestion
      {
        code: `
        const Component = () => {
          return <Child prop={async () => []} />;
        }`,
         output: `import { useCallback } from 'react';

        const Component = () => {
        const prop = useCallback(async () => [], []);
          return <Child prop={prop} />;
        }`,
        errors: [{ messageId: "function-usecallback-props" }],
      },
```

**Review by arthurgeron — APPROVED** [review]
LGTM. Thanks!

**Pull request merged**